cpp/src/io/json/write_json.cu

```diff
@@ -0,0 +1,733 @@
+    column_view const& column) const
+  {
+    return cudf::strings::detail::from_fixed_point(column, stream_, mr_);
+  }
+
+  // timestamps:
+  template <typename column_type>
+  std::enable_if_t<cudf::is_timestamp<column_type>(), std::unique_ptr<column>> operator()(
+    column_view const& column) const
+  {
+    std::string format = [&]() {
+      if (std::is_same_v<cudf::timestamp_s, column_type>) {
+        return std::string{"%Y-%m-%dT%H:%M:%SZ"};
+      } else if (std::is_same_v<cudf::timestamp_ms, column_type>) {
+        return std::string{"%Y-%m-%dT%H:%M:%S.%3fZ"};
+      } else if (std::is_same_v<cudf::timestamp_us, column_type>) {
+        return std::string{"%Y-%m-%dT%H:%M:%S.%6fZ"};
+      } else if (std::is_same_v<cudf::timestamp_ns, column_type>) {
+        return std::string{"%Y-%m-%dT%H:%M:%S.%9fZ"};
+      } else {
+        return std::string{"%Y-%m-%d"};
+      }
+    }();
+
+    // Since format uses ":", we need to add quotes to the format
+    format = "\"" + format + "\"";
+
+    return cudf::strings::detail::from_timestamps(
+      column,
+      format,
+      strings_column_view(column_view{data_type{type_id::STRING}, 0, nullptr}),
+      stream_,
+      mr_);
+  }
+
+  template <typename column_type>
+  std::enable_if_t<cudf::is_duration<column_type>(), std::unique_ptr<column>> operator()(
+    column_view const& column) const
+  {
+    auto duration_string = cudf::io::detail::csv::pandas_format_durations(column, stream_, mr_);
+    auto quotes = make_column_from_scalar(string_scalar{"\""}, column.size(), stream_, mr_);
+    return cudf::strings::detail::concatenate(
+      table_view{{quotes->view(), duration_string->view(), quotes->view()}},
+      string_scalar(""),
+      string_scalar("", false),
+      strings::separator_on_nulls::YES,
+      stream_,
+      mr_);
+  }
+
+  // lists:
+  template <typename column_type>
+  std::enable_if_t<std::is_same_v<column_type, cudf::list_view>, std::unique_ptr<column>>
+  operator()(column_view const& column, host_span<column_name_info const> children_names) const
+  {
+    auto child_view            = lists_column_view(column).get_sliced_child(stream_);
+    auto constexpr child_index = lists_column_view::child_column_index;
+    auto list_string           = [&]() {
+      auto child_string = [&]() {
+        if (child_view.type().id() == type_id::STRUCT) {
+          return (*this).template operator()<cudf::struct_view>(
+            child_view,
+            children_names.size() > child_index ? children_names[child_index].children
+                                                          : std::vector<column_name_info>{});
+        } else if (child_view.type().id() == type_id::LIST) {
+          return (*this).template operator()<cudf::list_view>(
+            child_view,
+            children_names.size() > child_index ? children_names[child_index].children
+                                                          : std::vector<column_name_info>{});
```

Comment:
Do we have tests where `children_names.size() <= child_index` ?

Reply:
added error test case for this scenario.